src/mibase.ts

```diff
@@ -136,50 +146,189 @@ export class MI2DebugSession extends DebugSession {
 	}
 
 	protected handleBreakpoint(info: MINode) {
+		let threadId = info.record("thread-id");
+
+		this.miDebugger.log("stdout", `Got theadid ${threadId}`);
+
+		let threadPid = this.threadToPid.get(parseInt(info.record("thread-id"), 10));
+
 		const event = new StoppedEvent("breakpoint", parseInt(info.record("thread-id")));
 		(event as DebugProtocol.StoppedEvent).body.allThreadsStopped = info.record("stopped-threads") === "all";
-		this.sendEvent(event);
+
+		this.miDebugger.log("stdout", `Handling breakpoint for threadPid == this.sessionPid (${threadPid} == ${this.sessionPid})`)
+
+		if (threadPid == this.sessionPid) {
+			this.sendEvent(event);
+		} else {
+			this.mi2Inferiors.forEach(inferior => {
+				if (threadPid == inferior.sessionPid) inferior.sendEvent(event)
+			});
+		}
```

Comment:
this construct is quite a lot of times, it should probably be moved into a utility function.

Also does this even respect the multiProcess configuration flag?

Reply:
I've refactored the way that mibase and miinferior works, allowed for much more reduction in the amount of code